### PR TITLE
change method name "withAddedRenderer -> withRenderer" and cleanup

### PR DESCRIPTION
**Pull request by thenotsoft**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

**Review by rustamwin — APPROVED** [review]

File: src/ErrorHandler/ErrorCatcher.php [hunk 1]
```diff
@@ -120,4 +114,15 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    private function validateMimeType(string $mimeType): string
+    {
+        if (strpos($mimeType, '/') === false) {
+            throw new \InvalidArgumentException('Invalid mime type.');
+        }
+        return strtolower($mimeType);
```
Comment (samdark) [inline src/ErrorHandler/ErrorCatcher.php:126]: Validation should not mutate the value.

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ErrorHandler/ErrorCatcher.php [hunk 2]
```diff
@@ -33,44 +33,39 @@ final class ErrorCatcher implements MiddlewareInterface
...
     }
 
-    public function withAddedRenderer(string $mimeType, string $rendererClass): self
+    public function withRenderer(string $mimeType, string $rendererClass): self
     {
-        if ($mimeType === '') {
-            throw new \InvalidArgumentException('The mime type cannot be an empty string!');
-        }
+        $mimeType = $this->validateMimeType($mimeType);
         if ($rendererClass === '') {
```
Comment (xepozz) [inline src/ErrorHandler/ErrorCatcher.php:49]: What about to check that class exists instead?

File: src/ErrorHandler/ErrorCatcher.php [hunk 3]
```diff
@@ -120,4 +121,19 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
+    private function validateMimeType(string $mimeType): void
+    {
+        if (strpos($mimeType, '/') === false) {
+            throw new \InvalidArgumentException('Invalid mime type.');
+        }
+    }
+
+    private function normalizeMimeType(string $mimeType): string
+    {
+        return strtolower($mimeType);
```
Comment (xepozz) [inline src/ErrorHandler/ErrorCatcher.php:137]:
```suggestion
        return strtolower(trim($mimeType));
```

**Review by xepozz — APPROVED** [review]

**Review by xepozz — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**